File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 07/08/2025, 19:32:26
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        maxProfit=0
        bestBuy=float('inf')
        for i in prices:
            maxProfit= max([maxProfit,i-bestBuy])
            bestBuy = min([bestBuy, i])
        return maxProfit
        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 07/08/2025, 19:32:20
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int bestBuy=prices[0],maxProfit=0;
        for(int i=1;i<prices.size();i++){
            maxProfit = max( maxProfit, prices[i]-bestBuy);
            bestBuy = min(bestBuy, prices[i]);
        }
        return maxProfit;
    }
};